### ISSUE TITLE
Add vmap-batched MCMC for same-structure models (same_structure=True)

### ISSUE THREAD

**Issue by amospagin** [amospagin/semla#55]
## Summary

When multiple Bayesian models share the same structure (same A/S/F pattern, same number of parameters) but use different data, we can use `jax.vmap` to evaluate all likelihoods in a single GPU kernel per NUTS step. This gives true GPU parallelism — 50 models in ~the same time as 1.

## Use case

Running many growth curve models (or CFA/SEM) with identical specification but different variable selections:

```python
results = batch_bayes(models, data, same_structure=True, gpu="auto")
```

## Design

1. Verify all models have identical parameter structure (same free/fixed pattern)
2. Stack observed data into a `(n_models, N, p)` tensor
3. Write a batched NumPyro model using `numpyro.plate("models", n_models)` where each plate element has its own parameters but shares the same compiled likelihood function
4. `jax.vmap` the likelihood computation across models — GPU evaluates all matrix inversions simultaneously
5. Run ONE MCMC with the batched model
6. Unpack per-model posterior samples into individual `BayesianResults`

## Why this works

- Same matrix dimensions → no padding needed
- Same free/fixed pattern → same parameter vector layout  
- Only data varies → perfect for vmap's `in_axes=(None, 0)` pattern
- Models are independent → block-diagonal structure, NUTS adapts well

## Key components

- Batched potential energy function (vmapped likelihood)
- Integration with `batch_bayes()` via `same_structure=True` parameter
- `BayesianResults.from_draws()` constructor for unpacking
- Verification that all models match in structure

## Expected speedup

- GPU vmap: ~1x single-model time regardless of batch size (until GPU memory saturates)
- CPU parallel: batch_size / n_cores × single-model time
- For 50 models on 6 CPU cores: ~8x speedup with GPU vmap

**Priority:** High (core workflow for the user)
